packages/client/tests/functional/issues/8832/tests.ts

```diff
@@ -0,0 +1,202 @@
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32767 ids', async () => {
+        const n = 32767
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32767 records', async () => {
+        const n = 32767
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 32768 ids', async () => {
+        const n = 32768
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 32768 records', async () => {
+        const n = 32768
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "in" has 65536 ids', async () => {
+        const n = 65536
+        const ids = await createTags(n)
+        const tags = await getTagsFindManyIn(ids)
+
+        expect(tags.length).toBe(n)
+      })
+
+      test('should succeed when "include" involves 65536 records', async () => {
+        const n = 65536
+        await createTags(n)
+        const tags = await getTagsFindManyInclude()
+
+        expect(tags.length).toBe(n)
+      })
+
+      /**
+       * Highlight scenarios of the query chunking logic behavior depending on how the bind variables are counted
+       * in the query filters.
+       */
+      describe('unhandled filters', () => {
+        test('should fail with `value too large to transmit` when "in" is repeated at least twice and "n" is $QUERY_BATCH_SIZE', async () => {
+          expect.assertions(3)
+          const n = 32766
+          const ids = await createTags(n)
+
+          try {
+            await prisma.tag.findMany({
+              where: {
+                OR: [
+                  {
+                    id: { in: ids },
+                  },
+                  {
+                    id: { in: ids },
+                  },
+                ],
+              },
+            })
+          } catch (e) {
+            const error = e as Error & { code: number; meta?: unknown }
+            expect(error.message).toContain(
+              'Assertion violation on the database: `too many bind variables in prepared statement, expected maximum of 32767, received 65533`',
+            )
+            expect(error.code).toBe('P2035')
+            expect(error.meta).toMatchObject({
+              database_error:
+                'too many bind variables in prepared statement, expected maximum of 32767, received 65533',
+            })
+          }
+        })
+
+        test('should fail with `value too large to transmit` when "in" has 32766 ids and a "take" filter', async () => {
```

Comment:
About "should fail with `value too large to transmit`" (applicable to all occurences)
I don't see `value too large to transmit` in the expects 🤔 
Is it part of the error message?

Reply:
I think that is actually outdated - the query never reaches tokio-postgres that would throw that error message, and instead we have something in quaint that catches it and responds with the error message that is expected below. We should probably update the description here.

Reply:
Thanks, the message is indeed outdated. Fixing it